src/parser/name_resolution.c

```diff
@@ -7152,6 +7167,82 @@ pt_str_compare (const char *p, const char *q, CASE_SENSITIVENESS case_flag)
     }
 }
 
+int
+pt_user_specified_name_compare (const char *p, const char *q)
+{
+  const char *dot_p = NULL;
+  const char *dot_q = NULL;
+  const char *original_p = NULL;
+  const char *original_q = NULL;
+
+  if (!p && !q)
+    {
+      return 0;
+    }
+
+  if (!p || !q)
+    {
+      return 1;
+    }
+
+  if (p[0] == '.' || q[0] == '.')
+    {
+      return 1;
+    }
+
+  dot_p = strchr (p, '.');
+  dot_q = strchr (q, '.');
+
+  if ((dot_p == NULL && dot_q != NULL) || (dot_p != NULL && dot_q == NULL))
+    {
+      /*
+       * In the case below, only after dot(.) is compared.
+       *
+       * e.g. p : user_name.object_name -> object_name
+       *      q : object_name           -> object_name
+       *
+       *      or
+       * 
+       *      p : object_name           -> object_name
+       *      q : user_name.object_name -> object_name
+       */
+      original_p = dot_p ? (dot_p + 1) : p;
+      original_q = dot_q ? (dot_q + 1) : q;
+
+      /*
+       * e.g. original_p : object_name.          -> NULL
+       *      original_q : user_name.object_name -> object_name
+       *
+       *      or
+       * 
+       *      original_p : user_name.object_name -> object_name
+       *      original_q : object_name.          -> NULL
+       */
+      if (!original_p || !original_q)
```

Comment:
`if (original_p[0] == '\0' ||  original_q[0] == '\0')`

Reply:
Okay, I'll change it. I should have checked the value.